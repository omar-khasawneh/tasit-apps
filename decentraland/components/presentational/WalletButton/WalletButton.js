import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import { responsiveHeight } from "react-native-responsive-dimensions";
import Button from "@presentational/Button";

// import { Linking } from "expo";
import AppLink from "react-native-app-link";

export default function WalletButton({ appName, scheme }) {
  console.info("appName", appName);
  console.info("scheme", scheme);
  const onConnect = async () => {
    try {
      // TODO: Pick scheme dynamically based on app name
      let url = `${scheme}://transaction?hello=world&goodbye=now`;

      console.info("Deep linking URL", url);

      // TODO: Change config to use test wallet app
      const config = {
        appName: appName,
        appStoreId: "1447390375", // TODO: Make dynamic
        appStoreLocale: "us",
        playStoreId: "io.gnosis.safe", // TODO: Make dynamic
      };

      console.info("App store config", config);

      // TODO: Query for presence of the app using a separate function
      await AppLink.maybeOpenURL(url, config);
    } catch (error) {
      // handle error
      console.info("Error", error);
    }
  };
  return (
    <View style={styles.buttonView}>
      <Button title="Connect with wallet" onPress={onConnect} />
    </View>
  );
}

WalletButton.propTypes = {
  appName: PropTypes.string.isRequired,
  scheme: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(5),
  },
});
