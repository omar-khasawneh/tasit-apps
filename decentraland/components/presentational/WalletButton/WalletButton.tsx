import React from "react";
import { StyleSheet, View, Platform, Linking } from "react-native";

import { responsiveHeight } from "react-native-responsive-dimensions";
import Button from "@presentational/Button";

export default function WalletButton({ appSlug, appName, scheme }) {
  // console.info("appSlug", appSlug);
  // console.info("appName", appName);
  // console.info("scheme", scheme);
  const openInStore = async ({
    appName,
    appStoreId,
    appStoreLocale = "us",
    playStoreId,
  }) => {
    try {
      // TODO: Pick scheme dynamically based on app name
      const url = `${scheme}://transaction?hello=world&goodbye=now`;

      console.info("Deep linking URL", url);
      // TODO: Move me outside of openInStore function
      // Show something on the screen dynamically based on the results of this
      // (that would need to happen outside of this component)
      const supported = await Linking.canOpenURL(url);
      if (!supported) {
        console.info("Can't handle url: " + url);
      } else {
        console.info("Able to handle url: " + url);
        // return Linking.openURL(url);
      }
    } catch (err) {
      console.info("An error occurred", err);
    }
    let appStoreUrl;
    if (Platform.OS === "ios") {
      appStoreUrl = `https://itunes.apple.com/${appStoreLocale}/app/${appName}/id${appStoreId}`;
      Linking.openURL(appStoreUrl);
    } else {
      appStoreUrl = `https://play.google.com/store/apps/details?id=${playStoreId}`;
      Linking.openURL(appStoreUrl);
    }
    console.info("app store url", appStoreUrl);
  };
  const onConnect = async () => {
    try {
      // TODO: Change config to use test wallet app
      const config = {
        appName: appSlug,
        appStoreId: "1447390375", // TODO: Make dynamic
        appStoreLocale: "us",
        playStoreId: "io.gnosis.safe", // TODO: Make dynamic
      };

      console.info("App store config", config);

      // TODO: Query for presence of the app using a separate function
      await openInStore(config);
    } catch (error) {
      // handle error
      console.info("Error", error);
    }
  };
  return (
    <View style={styles.buttonView}>
      <Button title={`Connect with ${appName}`} onPress={onConnect} />
    </View>
  );
}

// TODO: Migrate me to TypeScript types
WalletButton.propTypes = {
  appName: PropTypes.string.isRequired,
  appSlug: PropTypes.string.isRequired,
  scheme: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(5),
  },
});
