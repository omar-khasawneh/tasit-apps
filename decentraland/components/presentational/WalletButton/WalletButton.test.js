import React from "react";
import { shallow } from "enzyme";
import WalletButton from "@presentational/WalletButton";

describe("WalletButton", () => {
  it("renders the component", async () => {
    const appName = "testing";
    const scheme = "test-scheme";
    expect(
      shallow(<WalletButton appName={appName} scheme={scheme} />)
    ).toMatchSnapshot();
  });
});
