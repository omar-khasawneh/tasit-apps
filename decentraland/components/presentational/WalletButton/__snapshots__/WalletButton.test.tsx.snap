// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletButton renders the component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 66.7,
    }
  }
>
  <Button
    onPress={[Function]}
    title="Connect with testing"
  />
</View>
`;
